girder_worker/describe.py

```diff
@@ -0,0 +1,113 @@
+import six
+
+from inspect import getdoc
+try:
+    from inspect import signature
+except ImportError:
+    from funcsigs import signature
+
+
+class MissingDescriptionException(Exception):
+    """Raised when a task is missing description decorators."""
+
+
+class MissingInputException(Exception):
+    """Raised when a required input is missing."""
+
+
+def get_description_attribute(func):
+    """Get the private description attribute from a function."""
+    description = getattr(func, '_girder_description', None)
+    if description is None:
+        raise MissingDescriptionException('Task is missing description decorators')
+    return description
+
+
+def argument(name, type, *args, **kwargs):
```

Comment:
To be nit-picky, `type`, `input`, and `id` are NOT reserved words in python.  They are just references to built-in functions that can be redefined in any scope one likes.  Things like `and` and `as` are reserved words.  See Python [2](https://docs.python.org/2/reference/lexical_analysis.html#keywords) and [3](https://docs.python.org/3/reference/lexical_analysis.html#keywords) for a list of actual reserved words.